core/src/Cuda/Kokkos_Cuda_Half_Conversion.hpp

```diff
@@ -260,10 +260,16 @@ KOKKOS_INLINE_FUNCTION
 
 /************************** bhalf conversions *********************************/
 // Go in this branch if CUDA version is >= 11.0.0 and less than 11.1.0 or if the
-// architecture is not Ampere
+// architecture is older than Ampere
+#if !defined(KOKKOS_ARCH_KEPLER) && !defined(KOKKOS_ARCH_MAXWELL) && \
+    !defined(KOKKOS_ARCH_PASCAL) && !defined(KOKKOS_ARCH_VOLTA) &&   \
+    !defined(KOKKOS_ARCH_TURING75)
+#define KOKKOS_IMPL_ARCH_NVIDIA_GPU_AMPERE_PLUS
```

Comment:
Do we have other places in the code where we essentially need to test CUDA compute capability?

This is fine as a one-off internal flag, but we should consider whether something broader would be more helpful down the line.

Reply:
That's being addressed in #5948